src/lib/homestore_backend/replication_message.hpp

```diff
@@ -57,6 +80,49 @@ struct ReplicationMessageHeader {
     }
 };
 
+struct SyncMessageHeader : public BaseMessageHeader<SyncMessageHeader> {
+    SyncMessageHeader() : BaseMessageHeader() {
+        magic_num = HOMEOBJECT_RESYNC_MAGIC;
+        protocol_version = HOMEOBJECT_RESYNC_PROTOCOL_VERSION_V1;
+    }
+    SyncMessageType msg_type;
+    uint8_t reserved_pad[6]{};
+
+    bool corrupted() const{
+        if (magic_num != HOMEOBJECT_RESYNC_MAGIC || protocol_version != HOMEOBJECT_RESYNC_PROTOCOL_VERSION_V1) {
+            return true;
+        }
+        return BaseMessageHeader::corrupted();
+    }
+
+    std::string to_string() const {
+        return fmt::format(
+            "magic={:#x} version={} msg_type={} payload_size={} payload_crc={} header_crc={}\n",
+            magic_num, protocol_version, enum_name(msg_type), payload_size, payload_crc, header_crc);
+    }
+};
 #pragma pack()
 
+// objId is the logical offset of the snapshot in baseline resync
+// obj_id (64 bits) = type_bit (1 bit) | shard_id (48 bits) | batch_id (15 bits)
+// type_bit = 0 for HomeStore, 1 for HomeObject
+struct objId {
+    snp_obj_id_t value;
+    shard_id_t shardId;
+    snp_batch_id_t batchId;
+
+    objId(shard_id_t shard_id, snp_batch_id_t batch_id) : shardId(shard_id), batchId(batch_id) {
+        //type_bit (1 bit) | shard_id (48 bits) | batch_id (15 bits)
```

Comment:
please do overflow checking  before taking the 'or' operator.

i.e   if (shardid != shardid &0xffffffffffff)  // error

Reply:
this is to prevent overflow batchID /shardID corrupt other field

Reply:
Good point, will add checking for them.